game/screenshots.rpy

```diff
@@ -0,0 +1,196 @@
+        "I told you to ask if you wanted pictures!",
+        "Don't you listen?!{w=0.2} I said you should ask if you want pictures of me!"
+    ]
+
+    # Add the keymap for screenshots
+    jn_register_label_keymap("attempt_screenshot", "screenshot_dialogue", "s")
+
+# Attempt to produce a screenshot, render associated effects
+label take_screenshot:
+    hide window
+    python:
+        if persistent.affinity > -50:
+            renpy.screenshot("{0}/screenshot_{1}.png".format(_screenshot_dir, datetime.datetime.now().strftime(r"%d-%m-%Y_%H-%M-%S"))) # WIP; pictures will need to go to a sensible (local!) location
+            utils.log("Screenshot taken by player at {0}".format(datetime.datetime.now().strftime(r"%d/%m/%Y, %H:%M")))
+    play audio camera_shutter
+    with Fade(.15, 0, .50, color="#fff")
+    return
+
+# Handles dialogue and mechanics related to screenshots
+label screenshot_dialogue:
+    if _player_screenshot_in_progress:
+        # Don't take a screenshot if we're already going through the dialogue!
+        return
+    else:
+        $ _player_screenshot_in_progress = True
+
+    if persistent.jn_first_screenshot_taken is None:
+        # Set the date for the first ever screenshot, play the camera effects
+        $ persistent.jn_first_screenshot_taken = datetime.datetime.now()
+        call take_screenshot
+
+        n "H-huh?{w=0.2} What was that flash I just saw?"
+        n "Don't tell me...{w=0.3} was that a camera?!{w=0.2} There's a camera here?!"
+        n "..."
+        n "[player]...{w=0.3} d-did you do that...?"
+        menu:
+            "Yes, I did.":
+                n "O-oh!{w=0.2} Aha!{w=0.2} W-well,{w=0.1} at least you admit it."
+            "No, I didn't.":
+                n "Huh?{w=0.2} But then...{w=0.3} who...?"
+                n "..."
+            "I'm not sure.":
+                n "That's...{w=0.3} a little worrying..."
+                n "..."
+        n "Well, anyway.{w=0.1} The truth is,{w=0.1} I've never been very comfortable with having my picture taken without my permission."
+        n "I just...{w=0.3} really,{w=0.1} really don't like it."
+        n "So for the future,{w=0.1} could you please just let me know if you want to take pictures?"
+        n "I'd really appreciate it,{w=0.1} [player]."
+
+    # Positive screenshot route, as we have Natsuki's permission
+    elif player_screenshots_permission:
+        python:
+            # Update tracking and take the shot
+            persistent.jn_screenshot_good_shots_total += 1
+        n "Huh?{w=0.2} You're taking that picture now?"
+
+        if persistent.affinity >= 700:
+            n "Ahaha!{w=0.2} Sure!"
+            call take_screenshot
+        elif persistent.affinity < 700 and persistent.affinity > 300:
+            n "Well...{w=0.2} alright."
+            call take_screenshot
+        else:
+            n "...Fine.{w=0.1} Just be quick."
+            call take_screenshot
+
```

Comment:
We can simplify a little bit by just having the call after this if block since all of them do it
```suggestion
        if persistent.affinity >= 700:
            n "Ahaha!{w=0.2} Sure!"

        elif persistent.affinity < 700 and persistent.affinity > 300:
            n "Well...{w=0.2} alright."

        else:
            n "...Fine.{w=0.1} Just be quick."

        call take_screenshot

```